magpie/api/management/group/group_views.py

```diff
@@ -124,48 +117,31 @@ def delete_group_view(request):
-    Pending users are users who requested to join a group but still need to accept terms and conditions.
-    """
-    # Find the group associated with the request
-    group_name = ar.get_value_matchdict_checked(request, "group_name")
-    group = ax.evaluate_call(lambda: GroupService.by_group_name(group_name, db_session=request.db),
-                             fallback=lambda: request.db.rollback(), http_error=HTTPForbidden,
-                             msg_on_fail=s.GroupPendingUsers_GET_ForbiddenResponseSchema.description)
-    ax.verify_param(group, not_none=True, http_error=HTTPNotFound,
-                    msg_on_fail=s.Group_MatchDictCheck_NotFoundResponseSchema.description)
+    user_type = ar.get_multiformat_body(request, "user_type", default=UserGroupType.ACTIVE_USERGROUPS.value)
 
-    # Find all temporary tokens with requested group id that have a pending accept terms request
-    query = request.db.query(TemporaryToken).filter(TemporaryToken.group_id == group.id)
-    tmp_tokens = query.filter(TemporaryToken.operation == TokenOperation.GROUP_ACCEPT_TERMS)
-
-    # Find and return all user names associated with the discovered tokens
-    pending_user_names = [tmp_token.user.user_name for tmp_token in tmp_tokens]
-
-    # Remove any user already belonging to the group, in case any tokens are irrelevant.
-    # Should not happen since related tokens are deleted upon T&C acceptation.
+    user_names = []
     member_user_names = ax.evaluate_call(lambda: [user.user_name for user in group.users],
                                          http_error=HTTPForbidden,
                                          msg_on_fail=s.GroupUsers_GET_ForbiddenResponseSchema.description)
-    pending_user_names = [usr for usr in pending_user_names if usr not in member_user_names]
-
-    return ax.valid_http(http_success=HTTPOk, detail=s.GroupPendingUsers_GET_OkResponseSchema.description,
-                         content={"pending_user_names": pending_user_names})
+    if user_type in [UserGroupType.ACTIVE_USERGROUPS.value, UserGroupType.ALL_USERGROUPS.value]:
+        user_names += member_user_names
+    if user_type in [UserGroupType.PENDING_USERGROUPS.value, UserGroupType.ALL_USERGROUPS.value]:
+        # Find all temporary tokens with requested group id that have a pending accept terms request
+        tmp_tokens = request.db.query(TemporaryToken).filter(TemporaryToken.group_id == group.id)
+        tmp_tokens = tmp_tokens.filter(TemporaryToken.operation == TokenOperation.GROUP_ACCEPT_TERMS)
+
+        # Find and return all user names associated with the discovered tokens
+        pending_user_names = [tmp_token.user.user_name for tmp_token in tmp_tokens]
+
+        # Remove any user already belonging to the group, in case any tokens are irrelevant.
+        # Should not happen since related tokens are deleted upon T&C acceptation.
+        user_names += [usr for usr in pending_user_names if usr not in member_user_names]
```

Comment:
Uou can take advantage of sets to do this faster.
Define initial `user_names` as `set()` and `union` with anything to add. 
Then convert as back to list with `sorted(list(user_names))` call below.